### DISCUSSION TITLE
Can scripts create menu bar icons?

### DISCUSSION THREAD

**Discussion by kzvi** [johnlindquist/kit#860] [Ideas]
I would like to be able to write scripts that create a menu bar icon as a way of showing the status of a script / notifications to the user, for example:

- A script that periodically runs and checks for new email messages, and the menu bar icon for this script would change its icon if there are new messages. Clicking the menu bar icon could open gmail in a browser or could show a preview of the inbox.
- A script that syncs a folder to dropbox and changes its menu bar icon to show the progress and maybe to notify the user of errors if syncing fails.

**Comment by johnlindquist** [comment]
I've got a couple things in the works that you can try already:

## Status Icons

You can use `setStatus` that uses script kit icons (the next version will have better icons, still working on them):

```js
// Name: Testing Status

import "@johnlindquist/kit"

while (true) {
    await arg({
        placeholder: "Set status",
        onChoiceFocus: (_, { focused }) => { 
            let status = focused.value
            setStatus({
                status,
                message: `${status} message...`
            })
        }
    }, [
        "default",
        "success",
        "warn",
        "busy",
        "error",
        "paused"
    ])

    
}
```

[Open testing-status in Script Kit](https://scriptkit.com/api/new?name=testing-status&url=https://gist.githubusercontent.com/johnlindquist/890fd9a53104debe40365f4a2c928a5a/raw/ca5a163d2bc2ffea97a2de21691cdcccddfb49fc/testing-status.ts")

## Menu

The `menu` concept is still kind of a work in progress. I'm looking into making it a "second" menu from the main Kit.app menu, but for now it replaces it:

```js
// Name: Menu Example

import "@johnlindquist/kit"

// Set the icon to 👍 and let's you select scripts
menu(`👍`, [
    "app-launcher" //The name/id of the script you want to run
])
```

[Open menu-example in Script Kit](https://scriptkit.com/api/new?name=menu-example&url=https://gist.githubusercontent.com/johnlindquist/a85dad8e4e3a0756707c5ee9a4e10ead/raw/1d0b79facf4eeca7e74a7749240a7dbf37133975/menu-example.js")
